### PR TITLE
Add react chrome redux

### PR DESCRIPTION
**Pull request by kaglowka**
**todo...**:
- [x] ensure correct tabId within reducer in background
- [x] refactor reducer directories
- [x] improve convoluted thunk usage
- [x] explain why reference to `this.props.annotation` in DeleteAnnotationModal had to be artificially sustained.
- [x] debug tabs[null]
- [x] improve application initialization pipeline so it is more well-defined robust
- [x] refactor application initialization pipeline
- [ ] refactor selector usage after selector changes
- [x] make axios usage safe in all runtimes (now both CS and BG use it)
- [x] take into account jest mocking with the new store
- [x] remove `tabs` permission
- [x] update docs on all major issues
- [ ] work on error handling / propagation in the new (more divided) environment
- [x] fix tests
- [x] fix CI build
- [ ] add tests suited to the new environment
- [ ] work on extension reload scenario (browser port is invalidated and the content script usually breaks without reloading).

**nice to have**
- [ ] adapt redux-logger to react-chrome-redux dispatches
- [ ] adapt redux-logger to display current tab...

### CODE REVIEW

**Comment by kaglowka** [comment]
@glowka it is ready besides this weird circle ci failure no matter what ;) I'll give it one more try later but other than that PR is ready.

File: src/background/messages.ts [hunk 1]
```diff
@@ -28,3 +31,19 @@ export function setBadge(request, sender, sendResponse) {
...
+
+export function returnCurrentTabId(request, sender, sendResponse) {
+  if (request.action === 'GET_TAB_ID') {
+    if (sender.tab) {
+      // content script
+      console.debug('content script');
+      return sendResponse(sender.tab.id);
+    } else {
+      // popup
+      console.debug('popup');
```
Comment (glowka) [inline src/background/messages.ts]: console :)

File: src/background/background.ts [hunk 2]
```diff
@@ -52,6 +62,21 @@ function ppGaOnInstalled(details: InstalledDetails) {
   }
 }
 
+function getCookie(name: string): Promise<string | null> {
```
Comment (glowka) [inline src/background/background.ts]: Needs merging with https://github.com/PrzypisPowszechny/pp-client/blob/f0264e6e154fb91fde86154a853f1338a312d6d0/src/common/chrome-cookies.ts#L4

File: src/common/meta/script-type.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+enum ScriptType {
+  background,
```
Comment (glowka) [inline src/common/meta/script-type.ts]: :+1:

File: src/common/store/action-utils.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { markInThunkActionWithTabId, retrieveActionTab } from './tabs/action-tab';
+
+/*
+ * The primary actions (ordinary objects {} ) are marked by webext-redux with _sender.
+ * This allows to establish the tab for which the action was issued in the reducer
+ * Thunk actions, however, are functions issuing more functions within their body.
```
Comment (glowka) [inline src/common/store/action-utils.ts]: Some missing/redundant words in the sentence make it hard to get the sense of it.

File: src/common/store/action-utils.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * These must carry the tab information, too.
+ *
+ * This function converts a thunk into a tab-marked thunk by additionally decorating thunk (state, dispatch) arguments.
+ *
+ * "aliasAction" stands for the serializable action alias sent within browser message,
+ * mapped to a thunk on background page side.
+ */
+export function aliasActionToTabMarkedThunk(originalThunk) {
```
Comment (glowka) [inline src/common/store/action-utils.ts]: :+1:

File: src/content-scripts/components/viewer/report-editor/Suggestion.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { AnnotationAPIModel } from 'common/api/annotations';
 import { Reasons } from 'common/api/annotation-reports';
 import { PPScopeClass } from 'content-scripts/settings';
 import ppGa from 'common/pp-ga';
-import { AnnotationReportAPIModel, DataResponse } from '../../../../common/api/annotation-reports';
+import { AnnotationReportAPIModel, DataResponse } from 'common/api/annotation-reports';
```
Comment (glowka) [inline src/content-scripts/components/viewer/report-editor/Suggestion.tsx]: :+1:

File: src/common/api/redux-json-api-patch.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  /* eslint-disable */
+  loadNext = () => this.dispatch(readEndpoint(this.nextUrl));
+
+  loadPrev = () => this.dispatch(readEndpoint(this.prevUrl));
+  /* eslint-enable */
+}
+
+// PP: changed name
+export const readEndpointWithCustomOptions = (endpoint, {
+  options = {
```
Comment (glowka) [inline src/common/api/redux-json-api-patch.js]: :+1:

File: src/common/store/action-utils.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  return (aliasAction) => {
+    return (dispatch, getState) => {
+      const tabId = retrieveActionTab(aliasAction);
```
Comment (glowka) [inline src/common/store/action-utils.ts]: :+1:

**Pull request merged**